tree/dataframe/inc/ROOT/RDFHelpers.hxx

```diff
@@ -381,13 +385,11 @@ public:
 
       // We only print every n seconds.
       if (duration_cast<seconds>(system_clock::now() - fLastPrintTime) < fPrintInterval) {
-
          // Unless we are at the end of file processing, then we want to print the progress bar again (the final status)
          // Otherwise, if the last processed files are too small and they are processed in less than the time interval,
          // the final progress bar status would be incomplete. We want to prevent this from happening.
-
          if (fTotalFiles != currentFileIdx) {
-            if (currentFileIdx <= GetNEventsOfCurrentFile - fIncrement) {
+            if (GetNEventsOfCurrentFile - fProcessedEvents > fIncrement) {
```

Comment:
Can you comment on why we need the change in this logic here? Maybe also add the comment inline.

Reply:
the logic is kind of described in the comment above (but maybe I can improve it, it's true): if you print every second, at some point you process events too fast and you end up not really printing anymore so when you introduce the second statement, this is not true anymore. The change is due a bug in the implementation - I was comparing file index to the total number of events in a file instead of comparing the number of processed events to the total number of events (as it is done now). I also changed the format of the equation so it matches the one above for printing each second. Now, actually if we decide to anyways replace the final progress bar with the end state statement (and only print a one-liner), maybe this is not needed at all, let me test things and see what it looks like.

Reply:
actually now with the addition of the final status line, this trick is not needed anymore, I'll remove it